### ISSUE TITLE
"Sticky" module missing?

### ISSUE THREAD

**Issue by jffgrdnr** [angular/material.angular.io#473]
Hey folks, 

I pulled from your repo today and haven't been able to build because of a "sticky" module that isn't configured or imported? I think I can figure out how to fix it, but I don't know what module or NPM package its expecting?

![screen shot 2018-06-11 at 9 30 37 pm](https://user-images.githubusercontent.com/1308655/41268757-f8f34a58-6dbe-11e8-9491-95900cfdf74b.png)

**Comment by Chan4077** [comment]
If I'm right, it's supposed to be from the `@angular/cdk-experimental` package.

**Comment by JGardn-Quest** [comment]
From what I can tell, it's the **material-examples.umd.js** file that uses the "sticky" package. Do you have any idea what the syntax is for importing the "@angular/cdk-experimental" package into that file? The NPM page for cdk-experimental doesn't show how to import it. 

The material modules all have a "material.[module name]" syntax when being imported - but is cdk-experimental a material package?

![capture](https://user-images.githubusercontent.com/39810797/41301340-052c70f4-6e25-11e8-8164-b477986d790c.JPG)

**Comment by JGardn-Quest** [comment]
Sorry, I should have looked deeper before asking that question. The error in my first comment is cryptic, so I can't even tell which file is causing this issue, however, if it is the **material-examples.umd.js** file, that file is inside the @angular/material-examples node_module - so...

I could manually make sure that file references the missing module or schema, but since it's inside node, it's obviously not going to be a sustainable solution for me after I compile and try to push my code. Can you guys fix this issue?

**Comment by j-turner28** [comment]
I'm also seeing this same issue. I cloned the repo, ran `npm install` then ran `ng build` and got the error above. 

After some investigation it looks like the error is being caused by the changes in [this commit](https://github.com/angular/material2/commit/edbbc1bd7299df96a2f9c4200c3832d433f9cf5c#diff-f60f72def9d49393d2b325de8072565b) by @andrewseguin in the Material2 repo. 

In particular, the "table-sticky-complex-flex" and "table-sticky-complex" examples are throwing the error because of the "sticky" attribute not being known on ng-container.

**Comment by sebinbenjamin** [comment]
Same issue for me 

```ng serve --open
Your global Angular CLI version (6.0.8) is greater than your localversion (6.0.1). The local Angular CLI version is used.
To disable this warning use "ng config -g cli.warnings.versionMismatch false".** Angular Live Development Server is listening on localhost: 4200, open your browser on http://localhost:4200/ **
Date: 2018-06-16T05:45:22.962Z
Hash: 96ceff441dfd5e765a0f
Time: 23505ms
chunk {deeppurple-amber} deeppurple-amber.js (deeppurple-amber) 70.5 kB [initial] [rendered]
chunk {indigo-pink} indigo-pink.js (indigo-pink) 70.3 kB [initial] [rendered]
chunk {main} main.js (main) 1.94 kB [initial] [rendered]
chunk {pink-bluegrey} pink-bluegrey.js (pink-bluegrey) 72.4 kB [initial] [rendered]
chunk {purple-green} purple-green.js (purple-green) 72.4 kB [initial] [rendered]
chunk {runtime} runtime.js (runtime) 5.36 kB [entry] [rendered]
chunk {styles} styles.js (styles) 101 kB [initial] [rendered]
chunk {vendor} vendor.js (vendor) 325 kB [initial] [rendered]

ERROR in : Can't bind to 'sticky' since it isn't a known property of 'ng-container'.
1. If 'sticky' is an Angular directive, then add 'CommonModule' to the '@NgModule.imports' of this component.
2. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("-table [dataSource]="dataSource" *ngFor="let table of tables"><ng-container matColumnDef="position" [ERROR ->][sticky]="isSticky(stickyColumns, 'position')"><mat-header-cell *matHeaderCellDef>Position</mat-heade")
: Can't bind to 'sticky' since it isn't a known property of 'ng-container'.
1. If 'sticky' is an Angular directive, then add 'CommonModule' to the '@NgModule.imports' of this component.
2. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("*matFooterCellDef>Position Footer</mat-footer-cell></ng-container><ng-container matColumnDef="name" [ERROR ->][sticky]="isSticky(stickyColumns, 'name')"><mat-header-cell *matHeaderCellDef>Name</mat-header-cell><")
: Can't bind to 'stickyEnd' since it isn't a known property of 'ng-container'.
1. If 'stickyEnd' is an Angular directive, then add 'CommonModule' to the '@NgModule.imports' of this component.
2. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("l *matFooterCellDef>Name Footer</mat-footer-cell></ng-container><ng-container matColumnDef="weight" [ERROR ->][stickyEnd]="isSticky(stickyColumns, 'weight')"><mat-header-cell *matHeaderCellDef>Weight</mat-header")
: Can't bind to 'stickyEnd' since it isn't a known property of 'ng-container'.
1. If 'stickyEnd' is an Angular directive, then add 'CommonModule' to the '@NgModule.imports' of this component.
2. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("*matFooterCellDef>Weight Footer</mat-footer-cell></ng-container><ng-container matColumnDef="symbol" [ERROR ->][stickyEnd]="isSticky(stickyColumns, 'symbol')"><mat-header-cell *matHeaderCellDef>Symbol</mat-header")
: Can't bind to 'matHeaderRowDefSticky' since it isn't a known property of 'mat-header-row'.
1. If 'mat-header-row' is an Angular component and it has 'matHeaderRowDefSticky' input, then verify that it is part of this module.
2. If 'mat-header-row' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("at-footer-cell *matFooterCellDef>Filler footer cell</mat-footer-cell></ng-container><mat-header-row [ERROR ->]*matHeaderRowDef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-1')"></mat-header-row><ma")
: Property binding matHeaderRowDefSticky not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations". ("ell</mat-cell><mat-footer-cell *matFooterCellDef>Filler footer cell</mat-footer-cell></ng-container>[ERROR ->]<mat-header-row *matHeaderRowDef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-1')"></ma")
: Can't bind to 'matHeaderRowDefSticky' since it isn't a known property of 'mat-header-row'.
1. If 'mat-header-row' is an Angular component and it has 'matHeaderRowDefSticky' input, then verify that it is part of this module.
2. If 'mat-header-row' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("ef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-1')"></mat-header-row><mat-header-row [ERROR ->]*matHeaderRowDef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-2')"></mat-header-row><ma")
: Property binding matHeaderRowDefSticky not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations". ("w *matHeaderRowDef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-1')"></mat-header-row>[ERROR ->]<mat-header-row *matHeaderRowDef="displayedColumns; sticky: isSticky(stickyHeaders, 'header-2')"></ma")
: Can't bind to 'matFooterRowDefSticky' since it isn't a known property of 'mat-footer-row'.
1. If 'mat-footer-row' is an Angular component and it has 'matFooterRowDefSticky' input, then verify that it is part of this module.
2. If 'mat-footer-row' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("/mat-header-row><mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row><mat-footer-row [ERROR ->]*matFooterRowDef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-1')"></mat-footer-row><ma")
: Property binding matFooterRowDefSticky not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations". (", 'header-2')"></mat-header-row><mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>[ERROR ->]<mat-footer-row *matFooterRowDef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-1')"></ma")
: Can't bind to 'matFooterRowDefSticky' since it isn't a known property of 'mat-footer-row'.
1. If 'mat-footer-row' is an Angular component and it has 'matFooterRowDefSticky' input, then verify that it is part of this module.
2. If 'mat-footer-row' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component. ("ef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-1')"></mat-footer-row><mat-footer-row [ERROR ->]*matFooterRowDef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-2')"></mat-footer-row></m")
: Property binding matFooterRowDefSticky not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations". ("w *matFooterRowDef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-1')"></mat-footer-row>[ERROR ->]<mat-footer-row *matFooterRowDef="displayedColumns; sticky: isSticky(stickyFooters, 'footer-2')"></ma")

ℹ ｢wdm｣: Failed to compile.

```

**Comment by Nexeuz** [comment]
Same issue too, when apply **sticky** to mat-header-row in datatable, angular throw this error: 

```
[Angular] Property binding matHeaderRowDefSticky not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations".

[Angular] Can't bind to 'matHeaderRowDefSticky' since it isn't a known property of 'tr'.
```

`    <tr mat-header-row *matHeaderRowDef="displayedColumns; sticky: true"></tr>
`

**Comment by Nexeuz** [comment]
Fixed when it update material and cdk to lastest version.

**Comment by ShubhamJainSJ** [comment]
Which is the latest version that fixed the issue?

**Comment by amith009** [comment]
This issue fixed in '@angular/material": "^6.4.7'.

'<tr mat-header-row *matHeaderRowDef="displayedColumns; sticky:true"></tr>'

**Comment by jelbourn** [comment]
Closing since I think this is obsolete
